src/pyobs/icartt.py

```diff
@@ -97,7 +100,22 @@ def __init__ (self,Filenames,Alias=None,FixUnits=True,Verbose=False):
         # Navigation shorthands
         # ---------------------
         self._shorthands()
-                
+
+
+        # Subset for only_good point with valid lat and lon
+        # Needed for trj_sampler
+        # ----------------------------------------------------------
+        if only_good:
+            iGood = ~np.isnan(self.Nav['Longitude']) & ~np.isnan(self.Nav['Latitude']) 
+            for var in self.Vars:
+                self.__dict__[var] = self.__dict__[var][iGood]
```

Comment:
How will the code behave when there is no data with valid coordinates, that is the iGood array is empty?

Reply:
that is already implicitly taken care of.
iGood will always be an array of the same shape as self.Nav['Longitude'] - it will never be an empty array.
if all lons or all lats are invalid, the iGood array will be an array of all 'False' values.
If you subset a variable with an array of all False values it returns an empty array.

Reply:
Thanks for correcting me - indeed iGood is always non-zero length. Not quite clear on the second point, but I am left with the impression that returning an empty array is fine.